root/edit/details/AddRemoveAlias.js

```diff
@@ -0,0 +1,104 @@
+/*
+ * @flow
+ * Copyright (C) 2019 Anirudh Jain
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+import React from 'react';
+
+import bracketed from '../../static/scripts/common/utility/bracketed';
+import formatEntityTypeName from '../../static/scripts/common/utility/formatEntityTypeName';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import isDateEmpty from '../../static/scripts/common/utility/isDateEmpty';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import EntityLink from '../../static/scripts/common/components/EntityLink';
+import locales from '../../static/scripts/common/constants/locales';
+import isolateText from '../../static/scripts/common/utility/isolateText';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+
+type AliasEntityTypeT = $ElementType<AliasEntityTypeT, 'entityType'>;
+
+type DisplayDataT = {
+  alias: string,
+  begin_date: PartialDateT,
+  end_date: PartialDateT,
+  ended: boolean,
+  entity_type: AliasEntityTypeT,
+  locale: string,
+  primary_for_locale: boolean,
+  sort_name: string,
+  type: AliasT,
+};
+
+type Props = {
+  edit: {
+  ...EditT,
+  display_data: DisplayDataT,
+  },
+};
+
+const AddRemoveAlias = ({edit}: Props) => {
+  const display = edit.display_data;
+  const entityType = display.entity_type;
+  const entity = display[entityType];
+  return (
+    <table className={`details ${edit.edit_kind}-${entityType}-alias`}>
+      <tr>
+        <th>{addColonText(formatEntityTypeName(entityType))}</th>
+        <td>
+          <DescriptiveLink entity={entity} />
+          {' '}
+          {(entity && entity.gid) ? bracketed(<EntityLink content={l('view all aliases')} entity={entity} subPath="aliases" />) : null}
+        </td>
+      </tr>
+      <tr>
+        <th>{addColonText(l('Alias'))}</th>
+        <td>{isolateText(display.alias)}</td>
+      </tr>
+      {(display.sort_name === '') ? null : (
```

Comment:
There's a small bug here, since `sort_name` can be `null`. Just use `display.sort_name ?` (and invert the order of the expressions).